code/src/components/AddTodo.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import styled from 'styled-components'
+import Button from '@material-ui/core/Button'
+import TextField from '@material-ui/core/TextField'
+
+import { tasks } from '../reducers/tasks'
+
+const Form = styled.form`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  align-items: center;
+`
+const StyledButton = styled(Button)`
+  box-shadow: 0 4px 6px rgba(50, 50, 93, 0.11), 0 1px 3px rgba(0, 0, 0, 0.08);
+  padding: 7px 14px;
+  width: 100%;
+  height: 50px;
+  &:hover {
+    background-color: #5469d4;
+  }
+`
+const StyledTextField = styled(TextField)`
+  width: 100%;
+`
+
+export const AddTodo = () => {
+  const dispatch = useDispatch()
+  const [ text, setAddItem ] = useState('')
+  const [ id, setaddId ] = useState(0)
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    dispatch(
+      tasks.actions.addItem({
+        id, 
+        text,
+      })
+    )
+    setaddId(id + 1)
```

Comment:
This logic for the id works for now but it doesn't seem very sustainable because it depends on the component not being rerendered. As soon as you wanted to e.g. store your todos for a longer time, it wouldn't work anymore because it would always add the id 1 as soon as the page is reloaded. 
An alternative could be something like this, either here (then you need to use useSelector to get the information from the store) or inside your reducer (then you would need to adjust its functionality a little): 
`id: store.tasks.length === 0 ? 0 : Math.max(...store.tasks.map(item => item.id)) + 1,`
It looks for the largest id in your item-array and makes the id for the new item 1 larger than it. If the array is empty, the new id will be 0.

Reply:
Thank you for the feedback. I added a math floor with math random * 1024